pkgs/development/python-modules/isbnlib/default.nix

```diff
@@ -1,31 +1,55 @@
 {
   lib,
   buildPythonPackage,
-  fetchPypi,
-  nose,
-  coverage,
-  pythonOlder,
+  fetchFromGitHub,
+  pytestCheckHook,
+  pytest-cov,
+  setuptools,
 }:
 
 buildPythonPackage rec {
   pname = "isbnlib";
   version = "3.10.14";
-  format = "setuptools";
+  pyproject = true;
 
-  disabled = pythonOlder "3.7";
-
-  src = fetchPypi {
-    inherit pname version;
-    hash = "sha256-lvkIZMd7AfVfoR5b/Kn9kJUB2YQvO8cQ1Oq4UZXZBTk=";
+  src = fetchFromGitHub {
+    owner = "xlcnd";
+    repo = "isbnlib";
+    rev = "v${version}";
+    hash = "sha256-d6p0wv7kj+NOZJRE2rzQgb7PXv+E3tASIibYCjzCdx8=";
   };
 
+  build-system = [ setuptools ];
+
   nativeCheckInputs = [
-    nose
-    coverage
+    pytestCheckHook
+    pytest-cov
```

Comment:
Please patch out addopts instead.

Reply:
I think we should adopt a more seamless approach like https://github.com/NixOS/nixpkgs/pull/328550 if we don’t want to run coverage tests. Patching every package doesn’t scale.